security-analysis/src/main/java/com/powsybl/security/Security.java

```diff
@@ -219,14 +332,105 @@ public static String printLimitsViolations(List<LimitViolation> violations, Limi
+
+        List<LimitViolation> filteredLimitViolations = limitViolationFilter != null
+                ? limitViolationFilter.apply(result.getPreContingencyResult().getLimitViolations(), network)
+                : result.getPreContingencyResult().getLimitViolations();
+
+        try (TableFormatter formatter = formatterFactory.create(writer,
+                "Pre-contingency violations",
+                formatterConfig,
+                new Column(ACTION),
+                new Column(COUNTRY),
+                new Column(BASE_VOLTAGE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT),
+                new Column(EQUIPMENT + " (" + filteredLimitViolations.size() + ")"),
+                new Column(END),
+                new Column(VIOLATION_TYPE),
+                new Column(VIOLATION_NAME),
+                new Column(VALUE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(numberFormat),
+                new Column(LIMIT)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(numberFormat),
+                new Column(ABS_VALUE_LIMIT)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(numberFormat),
+                new Column(LOADING_RATE)
+                    .setHorizontalAlignment(HorizontalAlignment.RIGHT)
+                    .setNumberFormat(percentageFormat))) {
+            for (String action : result.getPreContingencyResult().getActionsTaken()) {
+                formatter.writeCell(action)
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell()
+                        .writeEmptyCell();
+            }
+            filteredLimitViolations.stream()
+                    .sorted(Comparator.comparing(LimitViolation::getSubjectId))
+                    .forEach(writeLinePreContingencyViolations(network, formatter));
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private static Consumer<? super LimitViolation> writeLinePreContingencyViolations(Network network, TableFormatter formatter) {
+        return violation -> {
+            try {
+                formatter.writeEmptyCell()
+                        .writeCell(LimitViolation.getCountry(violation, network) != null ? LimitViolation.getCountry(violation, network).name() : "")
+                        .writeCell(Float.isNaN(LimitViolation.getNominalVoltage(violation, network)) ? "" : Integer.toString((int) LimitViolation.getNominalVoltage(violation, network)))
```

Comment:
The return value of LimitViolation.getNominalVoltage(violation, network) cannot be NaN

Reply:
I've removed the test for NaN. Done.